app/code/core/Mage/Core/Model/Encryption.php

```diff
@@ -153,53 +148,52 @@ public function validateHashByVersion(#[\SensitiveParameter] $password, $hash, $
         return hash_equals($this->hash($salt . $password, $version), $hash);
     }
 
-    /**
-     * Instantiate crypt model
-     *
-     * @param string $key
-     * @return Varien_Crypt_Mcrypt
-     */
-    protected function _getCrypt($key = null)
+    public function encrypt(#[\SensitiveParameter] string $data): string
     {
-        if (!$this->_crypt) {
-            if ($key === null) {
-                $key = (string) Mage::getConfig()->getNode('global/crypt/key');
-            }
-            $this->_crypt = Varien_Crypt::factory()->init($key);
-        }
-        return $this->_crypt;
+        $key = Mage::getEncryptionKeyAsBinary();
+        $nonce = random_bytes(SODIUM_CRYPTO_SECRETBOX_NONCEBYTES);
+        $ciphertext = sodium_crypto_secretbox($data, $nonce, $key);
+        $encrypted = base64_encode($nonce . $ciphertext);
```

Comment:
```suggestion
        $encrypted = sodium_bin2base64($nonce . $ciphertext, SODIUM_BASE64_VARIANT_ORIGINAL);
```

Sorry deleted first suggestion here because not enough coffee yet. 😴 

From the PHP docs:

> Unlike base64_encode(), sodium_bin2base64() is constant-time (a property that is important for any code that touches cryptographic inputs, such as plaintexts or keys)

Reply:
phpstorm says they throw exception but I can't find anything in the doc and i can't make those functions throw exception... I don't think they throw (anymore?)

Reply:
ok for base64decode I managed to throw an exception so that part of code is ok, about the encode I wouldn't know what to do with an exception :-D

Reply:
Yeah, I'm not sure how you'd get `sodium_bin2base64` to throw either, and I wouldn't worry about catching it because as you said I'm not sure there's anything to do either.

Reply:
super, I've to check the last point of the checklist (which we could decide to use to alert backend to rotate the encryption key every once in a while?) and it should be done :-D